### PR TITLE
dogs now chase cats and cargo techs / qm

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Dogs now have handle automated movement changed up so they chase cats and cargo techs / qms.
66% of dogs will attack cats / techs, 33% will be playful.

## Why It's Good For The Game

"ooh puppy- **_AAAAAAAAAAAAAAAAAAAAAAAAAA_**"
> last words of some cargo tech.

Requested by marm, more small interactions are neat:tm:

## Testing

got brutally nuzzled and bit at by dogs.

## Changelog
:cl:
tweak: Dogs now chase cats and cargo techs / qms, and may bite them. Or just nuzzle them.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 1]
```diff
@@ -61,6 +71,73 @@
...
+
+		// No current target, or current target is out of range.
+		var/list/snack_range = oview(src, 3)
+		if(!movement_target || !(movement_target.loc in snack_range))
+			movement_target = null
+			stop_automated_movement = FALSE
+			var/mob/possible_target
+			for(var/mob/M in snack_range)
+				if(ishuman(M)) // mailman?
+					if(M.mind.assigned_role == "Cargo Technician" || M.mind.assigned_role == "Quartermaster") // MAILMAN!
```
Comment (lewcc) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:101]:
```suggestion
					if(M.mind && M.mind.assigned_role == "Cargo Technician" || M.mind.assigned_role == "Quartermaster") // MAILMAN!
```

**Comment by McRamon** [comment]
this is perfect

**Comment by tipoima** [comment]
>be QM
>HoP calls you to their office
>Ian tries to kill you
>Kill Ian in self-defense
>Get shot by HoP for killing Ian
>Get brigged

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 2]
```diff
@@ -61,6 +71,73 @@
...
+///This proc is used by dog subtypes to add on to their automated movements, so we don't get strange conflicts with ian down the line. Also lets ian be a good well trained perfect boy.
+/mob/living/simple_animal/pet/dog/proc/handle_dogomated_movement()
+	//Feeding, chasing food, FOOOOODDDD
+	if(IS_HORIZONTAL(src) || buckled)
+		return
+
+	if(++turns_since_scan > 5)
+		turns_since_scan = 0
+
+		// Has a target, but it's not where it was before, and it wasn't picked up by someone.
```
Comment (S34NW) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:88]: This comment is not quite true, specifically "but it's not where it was before"

Comment (Qwertytoforty) [reply code/modules/mob/living/simple_animal/friendly/dog.dm:88]: Let me tweak it, one moment

**Review by lewcc — COMMENTED** [review]
As noted, you should probably make sure this accounts for the movement target being qdeleted rather than just being nulled out. If that's already handled in automatic movement, then don't worry about it

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 3]
```diff
@@ -61,6 +71,73 @@
...
+
+		// No current target, or current target is out of range.
+		var/list/snack_range = oview(src, 3)
+		if(!movement_target || !(movement_target.loc in snack_range))
+			movement_target = null
+			stop_automated_movement = FALSE
+			var/mob/possible_target
+			for(var/mob/M in snack_range)
+				if(ishuman(M)) // mailman?
+					if(M.mind && M.mind.assigned_role == "Cargo Technician" || M.mind.assigned_role == "Quartermaster") // MAILMAN!
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/modules/mob/living/simple_animal/friendly/dog.dm:101]: I could, would require checking clothing slots on all human mobs in range as well as id card? Probably fine

**Comment by TravisAngeI** [comment]
Will the cats try to run away any?

Also I'm pretty sure this is just gonna get all the dogs killed the moment they step anywhere near a cargo tech ngl

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 4]
```diff
@@ -22,6 +22,16 @@
 	var/last_eaten = 0
 	footstep_type = FOOTSTEP_MOB_CLAW
 	var/next_spin_message = 0
+	var/turns_since_scan = 0
+	var/atom/movable/movement_target
+	var/mailman_intent = INTENT_HELP
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:27]: can you document this var please?

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (Contrabang) [reply code/modules/mob/living/simple_animal/friendly/dog.dm:101]: Just check their shown rank with (typecasted human) and `human.get_authentification_rank()` then check that rank against cargo tech and QM

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 5]
```diff
@@ -61,6 +71,73 @@
...
+	. = ..()
+	handle_dogomated_movement()
+
+///This proc is used by dog subtypes to add on to their automated movements, so we don't get strange conflicts with ian down the line. Also lets ian be a good well trained perfect boy.
+/mob/living/simple_animal/pet/dog/proc/handle_dogomated_movement()
+	//Feeding, chasing food, FOOOOODDDD
+	if(IS_HORIZONTAL(src) || buckled)
+		return
+
+	if(++turns_since_scan > 5)
```
Comment (DGamerL) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:85]: Is `++var` the same as `var++`?

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 6]
```diff
@@ -61,6 +71,73 @@
...
+						possible_target = M
+						a_intent = mailman_intent
+						break
+				else if(istype(M, /mob/living/simple_animal/pet/cat)) // CAT!
+					possible_target = M
+					a_intent = mailman_intent //Well trained dogs can be nice to cats. As a treat.
+					break
+			if(possible_target && (isturf(possible_target.loc) || ishuman(possible_target.loc))) // On the ground or in someone's hand.
+				movement_target = possible_target
+		if(movement_target)
```
Comment (DGamerL) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:111]:
```suggestion
		else
```

**Review by DGamerL — COMMENTED** [review]
Small stuff for now

**Comment by Qwertytoforty** [comment]
closing this one for now due to lack of motivation on this one in particular, sorry. Anyone is free to steal it

**Pull request closed without merging**